lib/mix/test/mix/tasks/xref_test.exs

```diff
@@ -863,6 +863,23 @@ defmodule Mix.Tasks.XrefTest do
                  "lib/b.ex"
                }
                """
+
+        assert Mix.Task.run("xref", ["graph", "--format", "json", "--output", "xref_graph.json"]) ==
+                 :ok
+
+        assert File.read!("xref_graph.json") ===
+                 String.trim_trailing("""
+                 {"lib/a.ex":{"lib/b.ex":"compile"},"lib/b.ex":{}}
+                 """)
+
+        assert Mix.Task.run("xref", ["graph", "--format", "json", "--output", "xref_graph.json"]) ==
```

Comment:
I would move the testing of the .bak functionality to mix/utils_test.exs, as it should be well encapsulated there.

Reply:
done. also added comments as to test directly i needed to make it public.